### PR TITLE
Base in home

### PR DESCRIPTION
**Pull request by sam-github**
connected to strongloop-internal/scrum-nodeops#640

This is based on #232, because it moved the sl-pm code around. I'll wait for the base PR to hit master before looking at the build state of this.

### CODE REVIEW

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Comment by kraman** [comment]
LGTM. 

Arc will need to be updated to pass a --base and --port variables or arc docs will need to mention that starting pm on the same machine requires setting those params

**Comment by sam-github** [comment]
@kraman arc doesn't use pm 4.x, so it doesn't need an update to use `--base`, though it would make it a bit more future proof in case they ever do upgrade (instead of just deleting the internal pm)

What's `--port`?

**Comment by kraman** [comment]
@sam-github `--port` is nothing, i meant `--listen` ... but i see arc already has `--listen 0`

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Comment by rmg** [comment]
Misread, thought `osenv` was a _new_ dependency.. FWIW, https://github.com/shama/userhome might be a closer fit and it has less dependencies:
```js
var home = require('userhome');
var base = home('.strong-pm');
// =>  /home/rmg/.strong-pm, /Users/rmg/.strong-pm, or C:\Users\rmg\.strong-pm
```

File: bin/sl-pm.js [hunk 1]
```diff
@@ -42,7 +43,7 @@ var parser = new Parser([
   ].join(''),
   argv);
 
-var base = '.strong-pm';
+var base = path.resolve(home('.strong-pm'));
```
Comment (rmg) [inline bin/sl-pm.js]: The `path.resolve` is done as part of `home()`.

**Comment by rmg** [comment]
Other than the extraneous call to `path.resolve`, LGTM.

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
ci refuses to build the head of this PR, I'm giving up.

**Pull request merged**